src/account.ts

```diff
@@ -413,18 +418,16 @@ export default class EthereumAccount {
 
         if (!receipt.status) {
           this.master._log.error(`Failed to deposit to channel: on-chain transaction reverted by the EVM`)
-          return Promise.reject(settlementBudget)
         } else {
           this.master._log.info(`Successfully deposited ${format(value, Unit.Wei)} to channel ${channelId} for account ${this.account.accountName}`)
         }
       } else {
         this.master._log.trace(`No on-chain funding transaction required to settle ${format(settlementBudget, Unit.Wei)} with account ${this.account.accountName}`)
       }
-
-      return settlementBudget
     } catch (err) {
       this.master._log.error(`Failed to fund outgoing channel to ${this.account.accountName}: ${err.message}`)
-      throw settlementBudget
+    } finally {
+      return settlementBudget
```

Comment:
I think it would be clearer if this `return` was in the `catch` with the log line, since if no error is thrown it will return from within the `try` block.

Reply:
Are you sure? Maybe I'm misinterpreting, but I think there are a number of cases where it wouldn't return the settlement budget from within the try (unless I added a return within both the try and catch).